sw/host/rom_ext_image_tools/signer/src/main.rs

```diff
@@ -71,128 +82,111 @@ fn parse_hex_str(hex_str: &str) -> Result<Vec<u8>> {
 
 /// Updates the manifest of an image.
 // TODO: This function must use a public key.
-fn update_image_manifest(
-    image: &mut Image,
-    key: impl AsRef<Path>,
-) -> Result<()> {
-    let key = fs::read(&key).with_context(|| {
-        format!("Failed to read the key from `{}`.", key.as_ref().display())
-    })?;
-    let key =
-        PrivateKey::parse_from_der(&key).context("Failed to parse the key.")?;
-
-    image.set_manifest_field(
-        &manifest::ROM_EXT_SIGNATURE_KEY_PUBLIC_EXPONENT,
-        key.public_exponent_be().iter().rev().cloned(),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_SIGNATURE_KEY_MODULUS,
-        key.public_modulus_be().iter().rev().cloned(),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_MANIFEST_IDENTIFIER,
-        parse_hex_str("0x4552544f")?,
-    )?;
-
-    image.set_manifest_field(
-        &manifest::ROM_EXT_IMAGE_VERSION,
-        // TODO: Use into_iter once it's stabilized (for all usages in this file).
-        std::array::IntoIter::new(0_u32.to_le_bytes()),
-    )?;
-
-    // TODO: Do we need these fields?
-    let extensions = [
-        (
-            &manifest::ROM_EXT_EXTENSION0_OFFSET,
-            &manifest::ROM_EXT_EXTENSION0_CHECKSUM,
-        ),
-        (
-            &manifest::ROM_EXT_EXTENSION1_OFFSET,
-            &manifest::ROM_EXT_EXTENSION1_CHECKSUM,
-        ),
-        (
-            &manifest::ROM_EXT_EXTENSION2_OFFSET,
-            &manifest::ROM_EXT_EXTENSION2_CHECKSUM,
-        ),
-        (
-            &manifest::ROM_EXT_EXTENSION3_OFFSET,
-            &manifest::ROM_EXT_EXTENSION3_CHECKSUM,
-        ),
-    ];
-    for fields in &extensions {
-        image.set_manifest_field(
-            fields.0,
-            std::array::IntoIter::new(0_u32.to_le_bytes()),
-        )?;
-        image.set_manifest_field(
-            fields.1,
-            std::array::IntoIter::new(0_u32.to_le_bytes()),
-        )?;
-    }
-    image.set_manifest_field(
-        &manifest::ROM_EXT_USAGE_CONSTRAINTS,
-        std::array::IntoIter::new(
-            [0; manifest::ROM_EXT_USAGE_CONSTRAINTS.size_bytes],
-        ),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO,
-        std::array::IntoIter::new(
-            [0; manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO.size_bytes],
-        ),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_IMAGE_TIMESTAMP,
-        std::array::IntoIter::new(0_u64.to_le_bytes()),
-    )?;
-    image.set_manifest_field(
-        &manifest::ROM_EXT_IMAGE_LENGTH,
-        std::array::IntoIter::new(u32::try_from(image.len())?.to_le_bytes()),
-    )?;
+fn update_image_manifest(image: &mut Image, key: &PrivateKey) -> Result<()> {
+    image.manifest.identifier = 0x4552544f;
```

Comment:
Would it perhaps make more sense to write
```
image.manifest = Manifest { ... };
```
with all the "obvious" fields, and tack on the fussy ones at the end? (or you can do them all at once with block expressions, I don't care as long as you're not repeating `image.manifest` like this is C89 =P)

Reply:
Done.